Description: Correctly determine the length of doc_root for its
 evaluation.
Origin: http://svn.php.net/viewvc?view=revision&revision=298277
Last-Update: 2010-05-02

Index: php/main/fopen_wrappers.c
===================================================================
--- php.orig/main/fopen_wrappers.c
+++ php/main/fopen_wrappers.c
@@ -430,8 +430,8 @@ PHPAPI int php_fopen_primary_script(zend
 		}
 	} else
 #endif
-	if (PG(doc_root) && path_info && (length = strlen(PG(doc_root)) &&
-		IS_ABSOLUTE_PATH(PG(doc_root), length))) {
+	if (PG(doc_root) && path_info && (length = strlen(PG(doc_root))) &&
+		IS_ABSOLUTE_PATH(PG(doc_root), length)) {
 		filename = emalloc(length + strlen(path_info) + 2);
 		if (filename) {
 			memcpy(filename, PG(doc_root), length);
